eng/pipelines/doc-index.yml

```diff
@@ -25,7 +25,6 @@ jobs:
             -RepoRoot $(Build.SourcesDirectory)
             -DocGenDir "$(Build.SourcesDirectory)/eng/docs/index"
             -verbose
-
       - task: UsePythonVersion@0
```

Comment:
Lets see about changing the output directory to somewhere outside of the SourcesDirectory, something like the artifact directory.

Reply:
We might also need to change the output path to the docfx tool as well so it isn't in the source tree.

Reply:
What about we put it `$(Build.BinariesDirectory)`